packages/blocks/src/api/factory.js

```diff
@@ -94,8 +95,8 @@ export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 		...block,
 		clientId,
 		attributes: {
-			...block.attributes,
-			...mergeAttributes,
+			...cloneDeep( block.attributes ),
```

Comment:
I don't expect this function would be called with too much frequency, but there's an optimization opportunity in that, if we are working with a cloned copy of the original attributes, we can be fine with the mutating behavior of [`Object.assign` on the target argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

```js
attributes: Object.assign(
	cloneDeep( block.attributes ),
	cloneDeep( mergeAttributes )
),
```

Where the implementation here would otherwise be more akin to:

```js
attributes: Object.assign(
	{},
	cloneDeep( block.attributes ),
	cloneDeep( mergeAttributes )
),
```

There's a small bit of overhead by copying properties of the cloned `block.attributes` into a fresh object.